### DISCUSSION TITLE
Fail to run dpgen after installation

### DISCUSSION THREAD

**Discussion by table9** [deepmodeling/dpgen#922] [Q&A]
After installation, the command "dpgen -h" shows the following error. Module pymatgen has already been installed (version 2022.8.23).

ModuleNotFoundError: No module named 'pymatgen.analysis.defects'

**Comment by table9** [comment]
Pymatgen version v2022.7.19 is the last version with pymatgen.analysis.defects module. An earlier version of pymatgen is needed.

**Comment by HuangJiameng** [comment]
![RE6rFKmqTS](https://user-images.githubusercontent.com/105633685/187406176-f1ccdbc0-7b29-4b7f-a523-d9ce4787ea15.jpg)

![JaOmhxoSUZ](https://user-images.githubusercontent.com/105633685/187405257-588bda3d-89c6-4855-b73e-6d36a98f0d7d.jpg)
Pymatgen v2022.7.19 will be the final release with the pymatgen.analysis.defects module included in the standard pymatgen package. It seems that there are two possible  solutions: 1) use Pymatgen of old version and 2)install `pymatgen-analysis-defects` in requirements. So what should we do? @njzjz @AnguseZhang @amcadmus

**Reply by AnguseZhang** [reply]
@kevinwenminion Please have a look, thanks.

**Comment by zcb-code** [comment]
I have the same problem, the conda version I have tried with Miniconda3-latest-Linux-x86_64.sh, Anaconda3-2022.05-Linux-x86_64.sh, Anaconda3-2021.04-Linux-x86_64.sh. The dp has been successfully installed. To install dpgen, I tried conda install -c conda-forge dpgen,
conda install -c conda-forge/label/cf202003 dpgen and conda install -c deepmodeling dpgen, pip install dpgen and downing install. It all not works. Except for that install dpgen with python3.6 environment, but have another problem with dpdata. Is that possible to write a note with recommended conda edition on readme text?

**Reply by HuangJiameng** [reply]
The DP-GEN on conda-forge is still verion 0.9.2. Please use `conda install -c deepmodeling dpgen` to install the latest version. see https://anaconda.org/deepmodeling/dpgen

**Comment by HuangJiameng** [comment]
After discussion, we suggest using `pip install pymatgen-analysis-defects` to install `defects` module if one's `pymatgen` is after version 2022.7.19. See https://github.com/materialsproject/pymatgen/releases/tag/v2022.7.19

**Reply by njzjz** [reply]
We should add it to `setup.py`.
